www/js/app/common/service/common-service.js

```diff
@@ -0,0 +1,44 @@
+var movieServices = angular.module('movieServices');
+movieServices.service('query', ['$resource', 'APP_CONSTANTS',  function($resource, APP_CONSTANTS) {
+	return {
+			'movieUpcoming' : $resource(APP_CONSTANTS.WS_URL +
+										APP_CONSTANTS.MOVIE_UPCOMING +
+										APP_CONSTANTS.API_KEY +
+										APP_CONSTANTS.PARAM_PAGE+':page'),
+
+	 		'genreMovieList': $resource(APP_CONSTANTS.WS_URL +
+										APP_CONSTANTS.GENRE_MOVIE_LIST +
+										APP_CONSTANTS.API_KEY)
+		   };
+}]);
+
+movieServices.factory('imageUrl', ['APP_CONSTANTS', function(APP_CONSTANTS) {
```

Comment:
Essencially to separate the RESTful endpoints from the rest of the common logic. However, I could have used factory instead of service and vice versa. 
when using a service you will get the instance of a function, when using a factory you will get the value that is returned by invoking the function reference.